src/custom-lists/background/storage.ts

```diff
@@ -0,0 +1,191 @@
+            query: {
+                id: { $in: listIds }
+            }
+        })
+        return this.changeListsbeforeSending(lists, pages)
+    }
+
+    // Function to insert into the DB
+    async insertCustomList({ name, isDeletable = 1, isNestable = 1 }: {
+        name: string, isDeletable: 0 | 1, isNestable: 0 | 1
+    }) {
+        return await this.storageManager.putObject(COLLECTION_NAME, {
+            name,
+            isDeletable,
+            isNestable,
+            createdAt: new Date(),
+        })
+    }
+
+    async updateListName({ id, name }: { id: Number, name: string }) {
+        await this.storageManager.updateObject(COLLECTION_NAME, {
+            id
+        },
+            {
+                $set: {
+                    name,
+                    createdAt: new Date(),
+                }
+            })
+    }
+
+    // Delete List from the DB.
+    async removeList({ id }: { id: number }) {
+        await this.storageManager.deleteObject(COLLECTION_NAME, {
+            id,
+        })
+        // Delete All pages associated with that list also
+        await this.storageManager.deleteObject(PAGE_LIST_ENTRY, {
+            listId: id,
+        })
+    }
+
+    // Adds mapping to lists and pages table.
+    async insertPageToList({ listId, pageUrl, fullUrl }:
+        { listId: number, pageUrl: string, fullUrl: string }) {
+
+        // check if the list ID exists in the DB, if not cannot add.
+        const idExists = Boolean(await this.fetchListById(listId))
+
+        if (idExists) {
+            await this.storageManager.putObject(PAGE_LIST_ENTRY, {
+                listId,
+                pageUrl,
+                fullUrl,
+                createdAt: new Date(),
+            })
+        }
+    }
+
+    // Removes the page from list.
+    async removePageFromList({ listId, pageUrl }: {
+        listId: number, pageUrl: number
+    }) {
+        const x = await this.storageManager.deleteObject(PAGE_LIST_ENTRY, {
+            $and: [
+                { listId: { $eq: listId } },
+                { pageUrl: { $eq: pageUrl } },
+            ],
+        })
+
+        return x
+    }
+
+    async checkPageInList({ listId, pageUrl }: { listId: Number, pageUrl: string }) {
+        const x = await this.storageManager.findObject(PAGE_LIST_ENTRY, {
+            $and: [
+                { listId: { $eq: listId } },
+                { pageUrl: { $eq: pageUrl } },
+            ],
+        })
+
+        return x
+    }
+
+    // Suggestions based on search in popup
+    async getListNameSuggestions({ name, url }: { name: string, url: string }) {
+        const lists = await this.storageManager.suggest(COLLECTION_NAME, {
+            name
+        })
+        const listIds = lists.map(({ id }: ListObject) => id)
+
+        // Gets all the pages associated with all the lists.
+        const pageEnteries = await this.storageManager.findAll(PAGE_LIST_ENTRY, {
+            listId: { $in: [...listIds] }, pageUrl: url,
```

Comment:
I don't think there's any need to create a new array here. i.e., the query `{ listId: { $in: listIds }, pageUrl: url }` _should_ work fine too (I know in my example I sent for this on Monday I created a new array, although that was more for demonstrative purposes to try and make it more obvious what's going on) Just a tiny thing.